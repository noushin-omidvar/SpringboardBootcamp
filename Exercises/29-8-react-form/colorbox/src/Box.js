import React from "react";
import "./Box.css";
const Box = ({ id, color, width, height, removeBox }) => {
  const boxStyle = {
    backgroundColor: color,
    width: Number(width),
    height: Number(height),
  };
  return (
    <div id={id} className="Box" style={boxStyle}>
      <button onClick={removeBox}> X </button>
    </div>
  );
};

export default Box;
