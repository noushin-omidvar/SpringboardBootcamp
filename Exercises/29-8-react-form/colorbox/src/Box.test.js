import { render, screen } from "@testing-library/react";
import Box from "./Box";

test("renders Box component", () => {
  const props = {
    id: "box-id",
    color: "blue",
    width: 200,
    height: 200,
    removeBox: jest.fn(),
  };
  const result = render(<Box {...props} />);

  // Smoke Test
  expect(screen.getByText("X")).toBeInTheDocument();

  // Snapshot Test
  const box = result.container.querySelector("#box-id");
  expect(box).toMatchSnapshot();
});
