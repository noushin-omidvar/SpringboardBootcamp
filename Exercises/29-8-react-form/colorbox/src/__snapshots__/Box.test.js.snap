// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Box component 1`] = `
<div
  class="Box"
  id="box-id"
  style="background-color: blue; width: 200px; height: 200px;"
>
  <button>
     X 
  </button>
</div>
`;
